core-processor/src/processing.rs

```diff
@@ -250,17 +282,22 @@ pub fn process_error(
         });
     }
 
-    let outcome = match dispatch.kind() {
-        DispatchKind::Init => DispatchOutcome::InitFailure {
-            program_id,
-            origin,
-            reason: err.to_string(),
-            executed,
-        },
-        _ => DispatchOutcome::MessageTrap {
-            program_id,
-            trap: err.to_string(),
-        },
+    let outcome = match case {
+        ProcessErrorCase::ExecutionFailed { .. } => {
+            let (_, err_payload) = case.to_reason_and_payload();
+            match dispatch.kind() {
+                DispatchKind::Init => DispatchOutcome::InitFailure {
+                    program_id,
+                    origin,
+                    reason: err_payload,
+                },
+                _ => DispatchOutcome::MessageTrap {
+                    program_id,
+                    trap: err_payload,
+                },
+            }
+        }
+        ProcessErrorCase::NonExecutable(_) => DispatchOutcome::NoExecution,
```

Comment:
so now for case when init message dies in waitlist we deposit event with NoExecution status instead of InitFailure?

Reply:
so we dont clean its waiting init queue, dont mark it inactive etc?

Reply:
we need test for that

Reply:
1) I did not change this logic
2) https://github.com/gear-tech/gear/blob/69cdfeab61f932c87e9e982333c524ba593754ef/pallets/gear/src/manager/task.rs#L177-L180